OpenRA.Mods.Common/Traits/AppearsOnRadar.cs

```diff
@@ -42,6 +42,14 @@ public void Created(Actor self)
 		{
 			var color = modifier != null ? modifier.RadarColorOverride(self) : self.Owner.Color.RGB;
```

Comment:
It would be better if we passed the color (either owner or team) into the modifier and then let these traits decide how best to deal with it.  The cloak trait could then apply the translucency, and the terrain color trait could check against UsePlayerStanceColors and do nothing if its set.